### PR TITLE
Some groundwork for Send

### PR DESCRIPTION
**Pull request by kspearrin**
This PR adds the groundwork to jslib for Send so that it can be implemented in other clients (starting with web, PR to come).

- Send APIs and tweaks to error parsing to support password required responses on unauthenticated send endpoints (access)
- Necessary models to parse API requests/responses
- Helpers to create parse send id and keys as URL-base-64 encoded strings
- Make send encryption keys by expanding with HKDF

### CODE REVIEW

File: src/models/view/sendAccessView.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    text = new SendTextView();
+    file = new SendFileView();
+
+    constructor(s?: SendAccess) {
+        if (!s) {
+            return;
+        }
+
+        this.id = s.id;
+        this.type = s.type;
```
Comment (cscharf) [inline src/models/view/sendAccessView.ts:22]: Wouldn't you also want to set `this.name = s.name`?

File: src/models/view/sendView.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    expirationDate: Date = null;
+    password: string = null;
+    disabled: boolean = false;
+
+    constructor(s?: Send) {
+        if (!s) {
+            return;
+        }
+
+        this.id = s.id;
```
Comment (cscharf) [inline src/models/view/sendView.ts:34]: Would you also want to set `this.name` and `this.notes` here as well?

File: src/models/view/sendAccessView.ts (hunk 1 above)
Comment (kspearrin) [reply src/models/view/sendAccessView.ts:22]: No, the pattern I followed has all non-encrypted things being set in the ctor like this while the encrypted values get decrypted and assigned in another function.

File: src/models/view/sendView.ts (hunk 2 above)
Comment (kspearrin) [reply src/models/view/sendView.ts:34]: same

**Review by cscharf — APPROVED** [review]

**Pull request merged**